### PR TITLE
Refresh io task topics

### PR DESCRIPTION
**Pull request by v-thepet**
## Summary

Refresh 
how-to-copy-directories
how-to-enumerate-directories-and-files

Part of User Story
https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1374049

### CODE REVIEW

**Comment by v-thepet** [comment]
@rpetrusha - Ready for your review per Task https://mseng.visualstudio.com/TechnicalContent/_queries/edit/1385009/?triage=true. Thanks.

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 1]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
-  
-### To enumerate file names in a directory and subdirectories  
-  
--   Use the <xref:System.IO.Directory.EnumerateFiles%28System.String%2CSystem.String%2CSystem.IO.SearchOption%29?displayProperty=nameWithType> method to search a directory and (optionally) its subdirectories, and to obtain a list of file names that match a specified search pattern.  
+|Directory names|<xref:System.IO.Directory.EnumerateDirectories%2A?displayProperty=nameWithType>|  
+|Directory information (<xref:System.IO.DirectoryInfo>)|<xref:System.IO.DirectoryInfo.EnumerateDirectories%2A?displayProperty=nameWithType>|  
+|File names|<xref:System.IO.Directory.EnumerateFiles%2A?displayProperty=nameWithType>|  
+|File information (<xref:System.IO.FileInfo>)|<xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType>|  
+|File system entry names|<xref:System.IO.Directory.EnumerateFileSystemEntries%2A?displayProperty=nameWithType>|  
+|File system entry information (<xref:System.IO.FileSystemInfo>)|<xref:System.IO.DirectoryInfo.EnumerateFileSystemInfos%2A?displayProperty=nameWithType>|
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]: One item to add:
|Directory and file names | \<xref:System.IO.Directory.EnumerateFileSystemEntries%2A?displayProperty=nameWithType>|

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 2]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
+The following example uses the <xref:System.IO.Directory.EnumerateDirectories%28System.String%29?displayProperty=nameWithType> method to get a list of the top-level directory names in a specified path.  
+
+[!code-csharp[System.IO.EnumDirs1#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.enumdirs1/cs/program.cs#1)]
+[!code-vb[System.IO.EnumDirs1#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.enumdirs1/vb/program.vb#1)]  
+
+The following example uses the <xref:System.IO.Directory.EnumerateFiles%28System.String%2CSystem.String%2CSystem.IO.SearchOption%29?displayProperty=nameWithType> method to recursively enumerate all file names in a directory and subdirectories that match a certain pattern. It then reads each line of each file and displays the lines that contain a specified string, with their filenames and paths.
+
+[!code-csharp[System.IO.Directory.EnumerateFiles#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directory.enumeratefiles/cs/program.cs#1)]
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]: in the example:
- in the VB example, there should be a blank line between the last `Imports` statement and the Module statement.
- the exception names should use camel casing or just be `ex`
- the 2 Console.WriteLine statements with format strings should be replaced with interpolated strings..

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 3]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
-### To enumerate a collection of DirectoryInfo objects  
+## Examples: Use the Directory class  
   
--   Use the <xref:System.IO.DirectoryInfo.EnumerateDirectories%2A?displayProperty=nameWithType> method to obtain a collection of top-level directories.  
-  
-     [!code-csharp[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/cs/program.cs#1)]
-     [!code-vb[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/vb/module1.vb#1)]  
+The following example uses the <xref:System.IO.Directory.EnumerateDirectories%28System.String%29?displayProperty=nameWithType> method to get a list of the top-level directory names in a specified path.  
+
+[!code-csharp[System.IO.EnumDirs1#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.enumdirs1/cs/program.cs#1)]
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]: in the example:
- delete `using System.Linq`;
- exception names should use camel casing. We generally just name an exception variable `ex`.
- we've been replaying format strings with interpolated strings; so:   
   - `$"{0}", dir.Substring(dir.LastIndexOf("\\") + 1)` --> `"{dir.Substring(dir.LastIndexOf("\\") + 1)}"`
   - "{0} directories found.",  dirs.Count --> $"{dirs.Count} directories found."

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 4]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
+[!code-csharp[System.IO.Directory.EnumerateFiles#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directory.enumeratefiles/cs/program.cs#1)]
+[!code-vb[System.IO.Directory.EnumerateFiles#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directory.enumeratefiles/vb/program.vb#1)]  
   
-### To enumerate a collection of FileInfo objects in all directories  
+## Examples: Use the DirectoryInfo class  
   
--   Use the <xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType> method to obtain a collection of files that match a specified search pattern in all directories. This example first enumerates the top-level directories to catch possible unauthorized access exceptions, and then enumerates the files.  
+The following example uses the <xref:System.IO.DirectoryInfo.EnumerateDirectories%2A?displayProperty=nameWithType> method to list a collection of top-level directories whose DateTime DirectoryInfo matches a certain pattern.  
+
+[!code-csharp[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/cs/program.cs)]
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]: - In the C# example, only the first two `using` statements are needed.
- The VB example should have a full example that corresponds to the C# example, not just a fragment
- indentation in the C# example should be fixed.
- See if the Snippet1 tag is used elsewhere in the dotnet/docs or dotnet/dotnet-api-docs repos (search for "directoryinfo.enumdirs#1"). If it's not, delete it.

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 5]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
+
+[!code-csharp[System.IO.Directory.EnumerateFiles#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directory.enumeratefiles/cs/program.cs#1)]
+[!code-vb[System.IO.Directory.EnumerateFiles#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directory.enumeratefiles/vb/program.vb#1)]  
   
-### To enumerate a collection of FileInfo objects in all directories  
+## Examples: Use the DirectoryInfo class  
   
--   Use the <xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType> method to obtain a collection of files that match a specified search pattern in all directories. This example first enumerates the top-level directories to catch possible unauthorized access exceptions, and then enumerates the files.  
+The following example uses the <xref:System.IO.DirectoryInfo.EnumerateDirectories%2A?displayProperty=nameWithType> method to list a collection of top-level directories whose DateTime DirectoryInfo matches a certain pattern.
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]:
```suggestion
The following example uses the <xref:System.IO.DirectoryInfo.EnumerateDirectories%2A?displayProperty=nameWithType> method to list a collection of top-level directories whose <xref:System.IO.DirectoryInfo.CreationTimeUtc> is earlier than a certain <xref:System.DateTime> value.  

```

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 6]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
+
+[!code-csharp[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/cs/program.cs)]
+[!code-vb[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/vb/module1.vb#1)]  
   
-     [!code-csharp[System.IO.DirectoryInfo.EnumerateDirectories#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumeratedirectories/cs/program.cs#1)]
-     [!code-vb[System.IO.DirectoryInfo.EnumerateDirectories#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directoryinfo.enumeratedirectories/vb/program.vb#1)]  
+The following example uses the <xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType> method to list all directories and all files whose Length FileInfo matches a certain pattern. This example first enumerates the top-level directories, to catch possible unauthorized access exceptions, and then enumerates the files.
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]:
```suggestion
The following example uses the <xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType> method to list all files whose <xref:System.IO.FileInfo.Length> exceeds 10MB. This example first enumerates the top-level directories to catch possible unauthorized access exceptions, and then enumerates the files.  
```

File: docs/standard/io/how-to-enumerate-directories-and-files.md [hunk 7]
```diff
@@ -11,54 +11,53 @@ ms.assetid: 86b69a08-3bfa-4e5f-b4e1-3b7cb8478215
...
+[!code-vb[System.IO.DirectoryInfo.EnumDirs#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directoryinfo.enumdirs/vb/module1.vb#1)]  
   
-     [!code-csharp[System.IO.DirectoryInfo.EnumerateDirectories#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumeratedirectories/cs/program.cs#1)]
-     [!code-vb[System.IO.DirectoryInfo.EnumerateDirectories#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.directoryinfo.enumeratedirectories/vb/program.vb#1)]  
+The following example uses the <xref:System.IO.DirectoryInfo.EnumerateFiles%2A?displayProperty=nameWithType> method to list all directories and all files whose Length FileInfo matches a certain pattern. This example first enumerates the top-level directories, to catch possible unauthorized access exceptions, and then enumerates the files.  
+
+[!code-csharp[System.IO.DirectoryInfo.EnumerateDirectories#1](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.io.directoryinfo.enumeratedirectories/cs/program.cs#1)]
```
Comment (rpetrusha) [inline docs/standard/io/how-to-enumerate-directories-and-files.md]: In the code examples:
- use interpolated strings except for the VB line that includes vbTab twice. That's best as a format string, with vbTab as one of the format items.
- in the VB example, delete `Imports System`
- exception variable names should be changed (camel cased or just `ex`)

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks, @v-thepet. Your changes look good. I've also suggested some additional changes, as well as some changes to modernize the code examples.

**Comment by v-thepet** [comment]
@rpetrusha - Changes are ready for review and in https://github.com/dotnet/samples/pull/574. Thanks.

**Comment by v-thepet** [comment]
@rpetrusha - Code changes are made in dotnet/samples#574.

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build after dotnet/samples#574 was merged.

**Comment by v-thepet** [comment]
@rpetrusha - Thanks for merging the code PR, this one should be ready for merge now. Thanks.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
This looks good, @v-thepet. I'll merge now.

**Pull request merged**